### PR TITLE
Add custom docker pipeline for multiarch builds

### PR DESCRIPTION
**Pull request by nkierpiec**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .expeditor/promote-docker-images.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#! /bin/bash
+
+export DOCKER_CLI_EXPERIMENTAL=enabled
+
+VERSION=$(cat VERSION)
```
Comment (jeremiahsnapp) [inline .expeditor/promote-docker-images.sh:5]: I don't think you want the contents of the VERSION file. Those contents could be different from the version that is being promoted. I think you can just have this script use `EXPEDITOR_VERSION`.

File: .expeditor/build-docker-images.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+docker build \
+  --build-arg "CHANNEL=${EXPEDITOR_CHANNEL}" \
+  --build-arg "VERSION=${EXPEDITOR_VERSION}" \
+  -t "chef/chef:${EXPEDITOR_VERSION}-${arch}" .
+
+echo "--- Pushing chef/chef:${EXPEDITOR_VERSION} docker image for ${arch} to dockerhub"
+docker push "chef/chef:${EXPEDITOR_VERSION}-${arch}"
+
+echo "--- Creating chef/chef:${EXPEDITOR_VERSION} multiarch manifest"
+docker manifest create "chef/chef:${EXPEDITOR_VERSION}" \
```
Comment (jeremiahsnapp) [inline .expeditor/build-docker-images.sh:17]: This `docker manifest create` command is only amending one architecture image in each build step and then that manifest is getting pushed in each build step. Does the final manifest somehow end up including both architecture images?

**Review by jeremiahsnapp — CHANGES_REQUESTED** [review]
This looks really awesome! I just have a couple comments.

File: .expeditor/promote-docker-images.sh (hunk 1 above)
Comment (nkierpiec) [reply .expeditor/promote-docker-images.sh:5]: Ah, yea - the whole head of promotion thing would make this a problem - good catch

File: .expeditor/build-docker-images.sh (hunk 2 above)
Comment (nkierpiec) [reply .expeditor/build-docker-images.sh:17]: Yes, this is exactly how it works - think of manifests as just a new tag and every amend just adds to that tag and updates it in place in dockerhub

**Review by jeremiahsnapp — APPROVED** [review]

File: .expeditor/promote-docker-images.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#! /bin/bash
+
+# EXPEDITOR_VERSION exists within the artifact_published workload
+
+export DOCKER_CLI_EXPERIMENTAL=enabled
+
+echo "--- Creating manifest for ${EXPEDITOR_TARGET_CHANNEL}"
+docker manifest create "chef/chef:${EXPEDITOR_TARGET_CHANNEL}" \
```
Comment (mimaslanka) [inline .expeditor/promote-docker-images.sh:8]: Suggest that we move this to a function
```
function create_and_push_manifest ()  {
     echo "---  Creating..."
     docker manifest create ${1}:${2}" \
         --amend ${1}:${EXPEDITOR_VERSION}-arm64 \
         --amend ....-amd64

     echo "--- Pushing...."
     docker manifest push....
}
create_and_push_manifest "chef/chef" "${EXPEDITOR_TARGET_CHANNEL}"
 
if ....
   create_and_pus_manifest "chef/chef" "latest"
  ...
```

**Pull request merged**